frontend/src/components/pagedItemsTable/PagedItemsTable.tsx

```diff
@@ -117,23 +118,109 @@ function PagedItemsTable<T extends object>({
 		[dispatch, setPageSizeCallback]
 	)
 
+	const shiftHeld = useRef(false)
+	useEffect(() => {
+		const handleKeyDown = (event: KeyboardEvent) => {
+			if (event.key === 'Shift') {
+				shiftHeld.current = true
+			}
+		}
+		const handleKeyUp = (event: KeyboardEvent) => {
+			if (event.key === 'Shift') {
+				shiftHeld.current = false
+			}
+		}
+		document.addEventListener('keydown', handleKeyDown)
+		document.addEventListener('keyup', handleKeyUp)
+		return () => {
+			document.removeEventListener('keydown', handleKeyDown)
+			document.removeEventListener('keyup', handleKeyUp)
+		}
+	}, [])
+
 	// We use this callback to respond when the user sorts a column
-	const sortByCallback: TableProps<any>['onChange'] = useCallback(
-		(pagination, filters, sorterResult) => {
-			if (!Array.isArray(sorterResult)) {
-				const sorter = sorterResult
-				const key = sorter.columnKey?.toString()
-				const order = sorter.order ?? undefined
-				if (key) {
-					if (sortBy === undefined || key !== sortBy.key || order !== sortBy.order) {
-						setSortByCallback({ key, order })
-					}
+	const sortByCallback = useCallback((columnKey: string) => {
+		const [existingFirst = undefined, existingSecond = undefined] = pagedItems.sortByList
+		const newFirst: Partial<SortBy> = { ...existingFirst }
+		const newSecond: Partial<SortBy> = { ...existingSecond }
+	
+		if (!shiftHeld.current) {
+			if (existingFirst?.key === columnKey) {
+				if (existingFirst.order !== 'descend') {
+					newFirst.order = 'descend'
+				} else {
+					newFirst.key = undefined
+					newFirst.order = undefined
 				}
+			} else if (existingFirst?.key !== columnKey) {
+				newFirst.key = columnKey
+				newFirst.order = 'ascend'
 			}
-		},
-		[sortBy, setSortByCallback]
-	)
-    const debouncedSortByCallback = useDebounce(sortByCallback)
+		} else {
+			if (newFirst.key === undefined) {
+				newFirst.key = existingFirst?.key
+				newFirst.order = existingFirst?.order
+			}
+			if (existingSecond?.key === columnKey) {
+				if (existingSecond.order !== 'descend') {
+					newSecond.order = 'descend'
+				} else {
+					newSecond.key = undefined
+					newSecond.order = undefined
+				}
+			} else if (existingSecond?.key !== columnKey) {
+				newSecond.key = columnKey
+				newSecond.order = 'ascend'
+			}
+		}
+	
+		if (newFirst.key === undefined || newFirst.key === newSecond.key) {
```

Comment:
might be simpler to fix this at the end when calling the back end ... you can just ignore undefined keys then... Less risk of falling out of sync or having our brain explode trying to track where are the sortBy gone.